### ISSUE TITLE
How do we clear up uncertainty around open source licenses?

### ISSUE THREAD

**Issue by rjmajma** [cfpb/project-open-source#6]
There are a lot of open source licenses out there. It can be confusing and difficult to figure out what they mean, how they impact the software you choose, and what can be done with subsequent modifications.

A decision tree, license primer, or other tool/resource would be very helpful for various components within an agency.

**Comment by willbarton** [comment]
I really like the way that Creative Commons handles clarifying their licenses, with [their "human readable" versions of "legal code"](http://creativecommons.org/licenses/by/4.0/) and iconography for specific categories of rights. It might beyond the scope of this effort to create human-readable descriptions for  something like OSI's [big list of licenses](http://opensource.org/licenses/category), but it should be doable for popular licenses.

It might also be useful, if we head down this path, to determine what major rights categories government would be most interested in, and then some iconography could be created around those.

**Comment by lheyman** [comment]
Does it make sense to try to come to a consensus on a recommended a preferred license for government developed projects? A recommendation like this would include a document explaining the rationalization for the choice, which would then also provide guidance and set of criteria for cases when one needs to (or needs to justify) using a license different from the recommended one.

The EU went so far as to write [their own license](https://joinup.ec.europa.eu/software/page/eupl/licence-eupl) for this purpose,  but that's probably far beyond what we need to do.

**Comment by Scotchester** [comment]
@lheyman :+1: on all counts.

In my opinion, we should recommend or even require that all new code developed by the government be CC0 in international jurisdictions (remembering that it is public domain by law in the U.S.), unless, as suggested, there is some kind of _really good_ justification for it not to be.

As to the questions of using and modifying _existing_ software, that is where a primer about the most common licenses could help agencies decide whether something is right for them to use/modify or not.

**Comment by lheyman** [comment]
Definitely.  Most of the primers/license breakdown/charts I've seen break down the different properties of each license. E.g. " here's a list of licences and a matrix of their comparative properties." Which is helpful when choosing a license, but doesn't seem super helpful in this scenario.

What do you think of coming at it in reverse? Namely, "if the software you're modifying is under license X, here are the things to think about/be concerned about."

**Comment by willbarton** [comment]
> What do you think of coming at it in reverse? Namely, "if the software you're modifying is under license X, here are the things to think about/be concerned about."

@lheyman I think that's where something like the Creative Commons "human-readable" formats and iconography would be:

a. Handy in its own right, and
b. Useful beyond the government context.

Particularly the iconography. If there were visuals that could provide a basic summary of popular languages, I think it'd be fantastically helpful for this particular scenario.
